patroni/postgresql/rewind.py

```diff
@@ -334,7 +347,13 @@ def execute(self, leader):
         #   waiting until Patroni on the master will expose checkpoint_after_promote=True
         checkpoint_status = leader.checkpoint_after_promote if isinstance(leader, Leader) else None
         if checkpoint_status is None:  # master still runs the old Patroni
-            leader_status = self._postgresql.checkpoint(self._conn_kwargs(leader, self._postgresql.config.superuser))
+            if self._postgresql.config.superuser['password'] is not None:
+                # we have superuser credentials, so run checkpoint on the leader
+                leader_status = self._postgresql.checkpoint(
+                        self._conn_kwargs(leader, self._postgresql.config.superuser))
+            else:
+                # query the leader whether a checkpoint happened using the rewind credentials
+                leader_status = self.check_leader_has_run_checkpoint(r)
```

Comment:
I think it is a bit more complex:
- With Postgres 10 and older, we don't have any other option rather than relying on superuser access.
- The `postgresql.authentication.superuser` could miss `username` and/or `password` or could be even totally absent from the config, but it doesn't mean that superuser access isn't possible, because libpq is using the local user by default. So the only way to figure out that we have superuser access is to try opening a connection with `self._postgresql.config.superuser` and if it doesn't work - switch to a waiting mode.

Reply:
To your first point, a valid (but painful for all but small clusters) option might be to re-clone; I think this is currently not being done no matter what, right? I think I tried both `remove_data_directory_on_rewind_failure` and `remove_data_directory_on_diverged_timelines` and it still failed not being able to do the Checkpoint. Is this a bug, and if so, is it worth fixing?

To your second point, fair enough. I already had a `have_superuser_credentials` method started, but then realized I don't need it for this PR (so far). Now I think it would make sense and might be of use elsewhere, what do you think?